app/service/rest.py

```diff
@@ -473,9 +473,12 @@ def cancel_notification_for_service(service_id, notification_id):
         raise InvalidRequest('Notification cannot be cancelled - only letters can be cancelled', status_code=400)
     elif not letter_can_be_cancelled(notification.status, notification.created_at):
         print_day = letter_print_day(notification.created_at)
-
+        if too_late_to_cancel_letter(notification.created_at):
+            message = "It’s too late to cancel this letter. Printing started {} at 5.30pm".format(print_day)
+        else:
+            message = "We could not cancel this letter. Probably it was already cancelled."
```

Comment:
We should avoid ambiguity in our log messages, as this creates more work for someone reading them: was it cancelled, or wasn't it?

How about we just print the status of the notification - "We could not cancel this letter. It has status X and was created at Y. [Optional: The letter processing deadline is Z.]" - so the reader can figure it out for themselves?

One advantage of this is that we don't need the utils changes, which helps keep the complexity down over here.

Reply:
Does the user see these error messages or are they only for our logs?

Reply:
Hey @quis that's actually a great question! The reason this error was ending up in our logs was because we were not handling it right in the admin. I made a PR now to handle it: https://github.com/alphagov/notifications-admin/pull/3984 ☺️ 

Hey @benthorner I like the idea of using notification status and date, I added it for the case when we do not know why the error happened. Since the errors will be user facing now (check above message for Chris), we will need to tell them exactly what happened, so I think making the change in utils makes sense. Especially since no new code was added in utils, I just moved the code for checking if it's too late to a separate method. So I'd say it's more of a little refactor rather than adding complexity ☺️